types/genesis.go

```diff
@@ -0,0 +1,116 @@
+package types
+
+import (
+	"crypto/sha512"
+	"fmt"
+
+	"github.com/algorand/go-algorand-sdk/encoding/msgpack"
+)
+
+// GenesisHashID is the Genesis HashID defined in go-algorand/protocol/hash.go
+var GenesisHashID = "GE"
+
+// A Genesis object defines an Algorand "universe"
+type Genesis struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	// The SchemaID allows nodes to store data specific to a particular
+	// universe
+	SchemaID string `codec:"id"`
+
+	// Network identifies the unique algorand network for which the ledger
+	// is valid.
+	Network string `codec:"network"`
+
+	// Proto is the consensus protocol in use at the genesis block.
+	Proto string `codec:"proto"`
+
+	// Allocation determines the initial accounts and their state.
+	Allocation []GenesisAllocation `codec:"alloc"`
+
+	// RewardsPool is the address of the rewards pool.
+	RewardsPool string `codec:"rwd"`
+
+	// FeeSink is the address of the fee sink.
+	FeeSink string `codec:"fees"`
+
+	// Timestamp for the genesis block
+	Timestamp int64 `codec:"timestamp"`
+}
```

Comment:
We need these too for parity with go-algorand.
```
Comment string codec:"comment"`

DevMode bool `codec:"devmode"`
```

Reply:
updated.